driver/src/main/scala/api/commands/writeresult.scala

```diff
@@ -32,8 +32,7 @@ sealed trait WriteResult {
 object WriteResult {
   def lastError(result: WriteResult): Option[LastError] = result match {
```

Comment:
Not sure there we lose any detail for bulk writes:

- With `ordered=true` mongo stop at first error, so using `LastError` (as in this PR) in a failed future seams ok since there is at most on error.
- With `ordered=false` a successful future of `MultiBulkWriteResult` is return in all cases and (should?) contains all errors.